src/main/java/com/powsybl/openloadflow/sensi/DcSensitivityAnalysis.java

```diff
@@ -7,297 +7,409 @@
+
+        lookForConnectivityLoss(lfNetwork, states, contingencies, contingenciesElements, equationSystem, straightforwardContingencies, losingConnectivityContingency);
+
+        Map<String, List<SensitivityValue>> contingenciesValue = new HashMap<>();
+        // compute the contingencies with no loss of connectivity
+        for (Contingency contingency : straightforwardContingencies) {
+            contingenciesValue.put(contingency.getId(), calculateSensitivityValues(lfNetwork, equationSystem, factorsGroups,
+                    states, functionReferenceByBranch, contingency.getElements().stream().map(element -> contingenciesElements.get(element.getId())).collect(Collectors.toList()),
+                    Collections.emptyMap()
+            ));
+        }
+
+        for (Map.Entry<List<Set<ComputedContingencyElement>>, List<Contingency>> contingencyEntry : losingConnectivityContingency.entrySet()) {
+            Map<SensitivityFactor, Double> predefinedResults = new HashMap<>();
+
+            contingencyEntry.getKey().stream().flatMap(Set::stream)
+                            .map(ComputedContingencyElement::getElement)
+                            .map(element -> lfNetwork.getBranchById(element.getId()))
+                            .forEach(lfBranch -> connectivity.getConnectivity().cut(lfBranch.getBus1(), lfBranch.getBus2()));
+
+            if (lfParameters.isDistributedSlack()) {
+                Map<Integer, List<ParticipatingElement>> participatingElementsPerCc = new HashMap<>();
+                participatingElements.forEach(element -> participatingElementsPerCc.computeIfAbsent(connectivity.getConnectivity().getComponentNumber(getParticipatingElementBus(element)), key -> new LinkedList<>())
+                                                                                   .add(element));
+                participatingElementsPerCc.values().forEach(ccElements -> ParticipatingElement.normalizeParticipationFactors(ccElements, "bus"));
+                Map<Integer, Map<String, Double>> participationPerCc = participatingElementsPerCc.entrySet().stream().collect(Collectors.toMap(
+                    Map.Entry::getKey,
+                    entry -> entry.getValue().stream().collect(Collectors.toMap(
+                        element -> getParticipatingElementBus(element).getId(),
+                        element -> -element.getFactor(),
+                        Double::sum
+                    ))
+                ));
+                getParticipationForBus = busId -> participationPerCc.get(connectivity.getConnectivity().getComponentNumber(lfNetwork.getBusById(busId)));
+                computeFactorsInjection(getParticipationForBus, factorsGroups, predefinedResults);
+
+                ComputedContingencyElement.setGlobalIndexes(contingenciesElements.values(), factorsGroups.size());
+                rhs = initRhs(lfNetwork, equationSystem, factorsGroups, contingenciesElements);
```

Comment:
Why do you solve multiple times (one for each list of element that loose connectivity) ?
Why you you keep in your rhs all elements and not only ones needed for contingencies that loose connectivity?

Reply:
We need to solve it multiple times, because the distribution of the slack changed, so we need to recompute the rhs for all the injection factors.
We actually do not need to recompute anything for the contingency elements, only for the factors, but once the LU decomposition is computed, I feel like the solve isn't really expensive, so I do not mind adding a few columns for nothing.

But you're right, we could optimize it by creating a big rhs with one column per combination (factor + elements responsible for the loss of connectivity) and solve everything at once

Reply:
The way solveTransposed is implemented does not allow reuse LU decomposition (but we could change it easily).
For solve, I did some test a few years ago and I think it was a little bit faster to group solve than calling solve multiple time (even if very fast in any cases so maybe not a bottleneck).

Reply:
> But you're right, we could optimize it by creating a big rhs with one column per combination (factor + elements responsible for the loss of connectivity) and solve everything at once

Bu are elements have already been computed before, we can just reuse it even for contingency that loose connectivity?

Reply:
Not sure I understand here. What exactly has already been computed, and can be reused ?

Reply:
elements responsible for the loss of connectivity